app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt

```diff
@@ -0,0 +1,28 @@
+package org.oppia.app.splash
+
+import androidx.fragment.app.testing.launchFragment
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+
+/** Tests for [SplashActivity]. */
+@RunWith(AndroidJUnit4::class)
+class SplashActivityTest {
```

Comment:
Another worthwhile test: making sure the HomeActivity is intented to. That apparently can be done with an additional Espresso library for intenting: https://stackoverflow.com/a/47517342/3689782.

Between that and the test below, those seem sufficient to me for the splash activity.

Reply:
but its not passing testcase. It is giving error saying `androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: with id: org.oppia.app:id/fragment_container`